app/user.go

```diff
@@ -288,7 +288,7 @@ func (a *App) createUserOrGuest(c *request.Context, user *model.User, guest bool
 		a.Srv().Go(func() {
 			pluginContext := pluginContext(c)
 			pluginsEnvironment.RunMultiPluginHook(func(hooks plugin.Hooks) bool {
-				hooks.UserHasBeenCreated(pluginContext, user)
+				hooks.UserHasBeenCreated(pluginContext, ruser)
```

Comment:
It's interesting to see that this never had a race condition. We are accessing a pointer in a separate goroutine here. Ideally, we should clone the user before sending it to the function. Anyways, out of scope for this issue.

Reply:
I think the reason we never had a race here because we don't write to that pointer after it passed here.